\chapter{Conclusion and Perspectives}
\label{chap:conclusion}

To summarize, the work presented in this thesis has been driven typically by the development and investigation of machine learning methods to address the computational challenges confronted in genomic data analysis: ranked-based approaches for improved breast cancer prognosis and network-guided approaches for enhanced biomarker discovery related to cancer survival. In fact, it is noteworthy that the theoretical and methodological contribution is significant and lies fundamentally in several branches of machine learning concerning applications across (but not limited to) cancer biology and social choice theory, specifically:


\paragraph{Learning with Rank Data.}

We have proposed two computationally attractive positive definite kernels between permutations, namely the Kendall and Mallows kernels, and further extended these kernels to rank data of complex structure that prevail in real-world applications including partial rankings, multivariate rankings and uncertain rankings (Chapter \ref{chap:kendall}). The significance of this work is of at least two folds: 
\begin{bulletList}
\item[1.] Thanks to these kernels, many kernel machines serve as off-the-shelf alternatives available to solve various problems pertaining to learning from rankings, or learning to rank, and can yield state-of-the-art performance that were demonstrated with an unsupervised cluster analysis of heterogeneous voting data and supervised biomedical classification tasks (Chapter \ref{chap:kendall}).
\item[2.] The Kendall embedding of the symmetric group brings forth novel incentives of learning on the symmetric group from unprecedented aspects. For instance, the Kendall embedding has motivated a geometric interpretation of the combinatorial problem of Kemeny aggregation based on which a tractable approximation bound was derived (Appendix \ref{chap:kemeny}) and can offer the opportunity of studying permutation problems such as seriation with yet another embedding following \cite{Fogel2013Convex, Lim2014Beyond}.
\end{bulletList}


\paragraph{Learning on Graphs.}

Given a network, we focused on network-guided feature selection coherent with the presumed network structure in two cases:
\begin{bulletList}
\item[1.] In case that the network is represented by an undirected graph and it encodes codependent relationships between features, assuming that neighboring features are encouraged to be selected simultaneously, we formalized the use of network-based wavelet smoothing as a regularization method for inducing structured sparsity with network-adaptive modularity in linear predictive models (Chapter \ref{chap:wavelet}).
\item[2.] In case that the network is represented by a directed graph and each circuit in it encodes the transduction of signal between features, assuming that circuit-level groups of features, if selected, are always selected simultaneously, we investigated the idea of first transforming the original representation of feature data into a circuit-level representation based on mathematical modeling of the network structure and then applying any standard feature selection algorithm which now becomes viable straightforwardly at the level of circuits (Chapter \ref{chap:hipathia}).
\end{bulletList}
Proof-of-methodology survival analysis of breast cancer was performed guided by a protein-protein interaction network (undirected) or a signaling pathway network (directed), and in both cases empirical superiority was demonstrated where biomarker discovery is enhanced by performing feature selection guided by a biological network as prior knowledge.


While the investigation of these machine learning topics covered in the present thesis and their applications in cancer prognosis is certainly unfinished as remarked in the discussion sections in each corresponding chapter, many interesting perspectives that were not covered in the present thesis remain to be explored. For example, while the thesis work concerns general prognosis for all breast cancer patients, there exist molecular subtypes of breast cancer based on specific genomic defects for which distinct prognostic tests or treatment strategies apply, and computational approaches such as unsupervised clustering or factor analysis \cite{Hastie2009Elements} can be used to stratify patients based on their genomic data beforehand, which would bring us one step further towards less costly and more effective personalized prognosis, personalized medicine and personalized treatment. Notably, all prognostic signatures elaborated in Section \ref{sec1:molecular} only apply to patients under specific clinico-pathological conditions. As another example, while the thesis work deals primarily with gene expression data, many other types of genomic data are available for analysis, among which DNA copy number variation (CNV) in array comparative genomic hybridization (aCGH) analysis and single-nucleotide polymorphism (SNP) in genome-wide association study (GWAS) are particularly widespread in active research in cancer biology, along with many other types of ``omics'' data including, to name just a few, epigenomics, proteomics, transcriptomics, metabolomics and microbiomics, and standard clinico-pathological parameters which incontrovertibly still dominate clinical practice of breast cancer prognosis until today. To make use of multi-omics data in an integrative analysis, multi-view learning \cite{Sun2013survey} is such a branch in machine learning that studies how to combine different and heterogeneous views of a sample. In particular, within the paradigm of kernel learning, if a kernel is defined for each view of the data, multiple kernel learning \cite{Goenen2011Multiple} has been shown relevant for genomic data fusion \cite{Lanckriet2004statistical}.


One thing that needs to be explicitly pointed out is that the contribution of the thesis work to genomic data analysis for breast cancer prognosis has been mainly theoretical and methodological with efforts to propose new ideas of improving prognosis and designing biomarkers, but the numerical results are far from reaching clinical significance. In particular, we do not claim to have identified any multigene signature for breast cancer prognosis, and we treat the lists of prognostic genes or biomarkers identified from our studies with certain extent of skepticism. A somehow discouraging fact in the field of computational cancer research is that a voluminous literature of more than 150,000 papers documenting thousands of claimed biomarkers has been produced in medicine of which fewer than 100 have been validated for routine clinical practice \cite{Poste2011Bring}, and even fewer than 20 are recognized with variable levels of evidence in the 2014 European Society of Medical Oncology (ESMO) clinical practice guidelines for lung, breast, colon and prostate cancer \cite{Schneider2015Establishing}. Compared to the number of research findings in this area, the very few number of gene expression-based breast cancer prognostic signatures successfully implemented in clinical routines (Section \ref{sec1:molecular}) has inevitably raised controversies on the practical validity of molecular signatures. This is mainly because the vast majority of those findings are deficient in a proper validation procedure, not to mention validation oriented for clinical implication, resulting in an exaggeration of trivial findings and their clinical utility so that a large number of claimed signatures could very likely fail to add significantly incremental values assisting prognosis assessment and therapeutic decision making in addition to the use of standard clinico-pathological parameters. Notwithstanding, the research community generally holds an optimistic prospective towards the future, as long as proper validation pipelines will be taken systematically in all forthcoming research \cite{Michiels2016Statistical}. Since objectives accounting for clinical utility were not at all entailed in the first place and neither meta-analysis-based validation with multiple datasets nor cross-study validation was performed during the course of my doctoral studies, we cannot conclude with any convincing prognostic signatures. However, we will try our best to venture some caveats suggesting pitfalls in analyzing genomic data specifically for biomarker discovery in cancer prognosis, in line with many previous attempts \cite{Ambroise2002Selection, Simon2003Pitfalls, Issaq2011Cancer, Weigelt2012Challenges}:


\paragraph{Should We Engage in Biomarker Discovery?} Regarding prognosis improvement, currently held belief rationalizing biomarker discovery by the research community is that a prognostic model based on only a few biomarkers should better capture and explain the biological complexity related to cancer survival. However, an anti-intuitive yet intriguing phenomenon already observed by previous studies \cite{Haury2011influence} arises that inference based on a few selected ones compared to the use of all features available covering a much larger spectrum of genome often does \emph{not} lead to drastic improvement and sometimes even lead to slight deterioration, as reported across all three studies in the present thesis: in Section \ref{sec2:classification} it did not seem to be beneficial to perform feature selection in the biomedical applications thereof when Support Vector Machines with the Kendall kernel is used to classify genomic profiles, in Section \ref{sec3:breastcancer} the best-performing model with the highest accuracy of survival risk prediction was the simplest ridge regression in which no feature selection was carried out, in Section \ref{sec4:performance} the most frequently selected or best suited classifiers for all predictive profiles in the breast cancer prognosis classification task are Support Vector Machines with various kernels none of which involve feature selection. Fortunately, model performance did not degrade significantly when the initiative of performing feature selection is supplemented in the predictive models at least in the latter two cases. In particular, another disagreeably striking observation is that in the benchmark study in Section \ref{sec3:breastcancer}, despite that they indeed show superiority with respect to several evaluation criteria of feature selection efficacy such as stability, connectivity and interpretability, all tested network-guided feature selection methods performed worse in terms of survival risk prediction, albeit insignificantly, than the simple and network-free elastic net. In a nutshell, it is not supported by existing evidence that feature selection in genomic data analysis could guarantee to make prognosis more accurate, and this convention is merely an assumption still awaiting to be confirmed, which therefore should be taken cautiously.


\paragraph{Should We Trust the Biomarkers Discovered?} The trust we should invest in the biological values of the biomarkers discovered from cancer research can be limited by many factors and thus their merits in the development of clinical prognostic assays (or to further suggest therapeutic targets) should be taken with extreme caution. One issue regarding insignificant improvement of prognosis accuracy due to feature selection has been discussed in the last paragraph. Other factors that can influence the validity of feature selection and should also be considered as indispensable requirements for the selected features to be considered candidate biomarkers of interest include cross-study reproducibility and functional interpretability of the identified biomarkers. Unfortunately, these issues are indeed demanding challenges. It is rarely the case that two prognostic signatures identified with different analytical methods and/or based on different datasets have a significant overlap, for instance only three genes are in common in the now famous 70-gene signature of \cite{Veer2002Gene} versus the 76-gene signature of \cite{Wang2005Gene}. Even surprisingly, \cite{Venet2011Most} reported that most random gene expression signatures are significantly associated with breast cancer outcome, criticizing on a hypothesis that the performance of prognostic models using deliberately selected features can be within the range of likely values based on random selection of features. Several studies analyzed the difficulty in selecting robust signatures, and overall concluded that the lack of robustness is mainly due to the fact that many different sets of genes with little overlap can nonetheless collectively have similar predictive power and the situation should be expected to be ameliorated when in the future we can gather a much larger number of samples to draw conclusion on \cite{Michiels2005Prediction, Ein-Dor2006Thousands, Haury2011influence}. A major drawback is that nowadays numerous discoveries that are based on small and unrepresentative datasets hardly sustain independent validation so that their clinical utility remains out of reach. In particular, the numerical results presented in this thesis requires cross-study validation too as already mentioned above, subject to many impending issues arising in cross-study validation such as test set bias that could affect reproducibility and needs meticulous attention as well \cite{Patil2015Test}.


\section*{Some Last Words...}

Our knowledge and understanding of cancer biology is still far incomplete but we are given the extraordinary opportunity in the era of ``omics'' revolution and data science to study cancer with machine learning. Just bear in mind that opportunities come with caveats that it necessarily calls for comprehensive study and proper validation as well as concerns such as clinical utility and cost-effectiveness of the computational findings on the road to breakthrough discoveries and success in the fight against cancer.
