\chapter{Conclusion and Perspectives}
\label{chap:conclusion}

To summarize, the work presented in this thesis has been driven typically by the development and investigation of machine learning methods to address the computational challenges confronted in genomic data analysis: ranked-based approaches for improved breast cancer prognosis and network-guided approaches for enhanced biomarker discovery related to cancer survival. In fact, it is noteworthy that the theoretical and methodological contribution is significant and lies fundamentally in several branches of machine learning concerning applications across (but not limited to) cancer biology and social choice theory, specifically:


\paragraph{Learning with Rank Data.}

We have proposed two computationally attractive positive definite kernels between permutations, namely the Kendall and Mallows kernels, and further extended these kernels to rank data of complex structure that prevail in real-world applications including partial rankings, multivariate rankings and uncertain rankings (Chapter \ref{chap:kendall}). The significance of this work is of at least two folds: 
\begin{bulletList}
\item[1.] Thanks to these kernels, many kernel machines serve as off-the-shelf alternatives available to solve various problems pertaining to learning from rankings, or learning to rank, and can yield state-of-the-art performance that were demonstrated with an unsupervised cluster analysis of heterogeneous voting data and supervised biomedical classification tasks (Chapter \ref{chap:kendall}).
\item[2.] The Kendall embedding of the symmetric group brings forth novel incentives of learning on the symmetric group from unprecedented aspects. For instance, the Kendall embedding has motivated a geometric interpretation of the combinatorial problem of Kemeny aggregation based on which a tractable approximation bound was derived (Appendix \ref{chap:kemeny}) and can offer the opportunity of studying permutation problems such as seriation with yet another embedding following \cite{Fogel2013Convex, Lim2014Beyond}.
\end{bulletList}


\paragraph{Learning on Graphs.}

Given a network, we focused on network-guided feature selection coherent with the presumed network structure in two cases:
\begin{bulletList}
\item[1.] In case that the network is represented by an undirected graph and it encodes codependent relationships between features, assuming that neighboring features are encouraged to be selected simultaneously, we formalized the use of network-based wavelet smoothing as a regularization method for inducing structured sparsity with network-adaptive modularity in linear predictive models (Chapter \ref{chap:wavelet}).
\item[2.] In case that the network is represented by a directed graph and each circuit in it encodes the transduction of signal between features, assuming that circuit-level groups of features, if selected, are always selected simultaneously, we investigated the idea of first transforming the original representation of feature data into a circuit-level representation based on mathematical modeling of the network structure and then applying any standard feature selection algorithm which now becomes viable straightforwardly at the level of circuits (Chapter \ref{chap:hipathia}).
\end{bulletList}
Proof-of-methodology survival analysis of breast cancer was performed guided by a protein-protein interaction network (undirected) or a signaling pathway network (directed), and in both cases empirical superiority was demonstrated where biomarker discovery is enhanced by performing feature selection guided by a biological network as prior knowledge.


While the investigation of these machine learning topics covered in the present thesis and their applications in cancer prognosis is certainly unfinished as remarked in the discussion sections in each corresponding chapter, many interesting perspectives that were not covered in the present thesis remain to be explored. For example, while the thesis work concerns general prognosis for all breast cancer patients, there exist molecular subtypes of breast cancer based on specific genomic defects for which distinct prognostic tests or treatment strategies apply, and computational approaches such as unsupervised clustering or factor analysis \cite{Hastie2009Elements} can be used to stratify patients based on their genomic data beforehand, which would bring us one step further towards less costly and more effective personalized prognosis, personalized medicine and personalized treatment. Notably, all prognostic signatures elaborated in Section \ref{sec1:molecular} only apply to patients under specific clinico-pathological conditions. As another example, while the thesis work deals primarily with gene expression data, many other types of genomic data are available for analysis, among which DNA copy number variation (CNV) in array comparative genomic hybridization (aCGH) analysis and single-nucleotide polymorphism (SNP) in genome-wide association study (GWAS) are particularly widespread in active research in cancer biology, along with many other types of ``omics'' data including, to name just a few, epigenomics, proteomics, transcriptomics, metabolomics and microbiomics, and standard clinico-pathological parameters which incontrovertibly still dominate clinical practice of breast cancer prognosis until today. To make use of multi-omics data in an integrative analysis, multi-view learning \cite{Sun2013survey} is such a branch in machine learning that studies how to combine different and heterogeneous views of a sample. In particular, within the paradigm of kernel learning, if a kernel is defined for each view of the data, multiple kernel learning \cite{Goenen2011Multiple} has been shown relevant for genomic data fusion \cite{Lanckriet2004statistical}.


One thing that needs to be explicitly pointed out is that the contribution of the thesis work to genomic data analysis for breast cancer prognosis has been mainly theoretical and methodological with an effort to propose new ideas of improving prognosis and designing biomarkers, but the numerical results are far from reaching clinical significance. In particular, we do not claim to have identified any multigene signature for breast cancer prognosis, and we treat the lists of prognostic genes or biomarkers identified from our studies with certain extent of skepticism. A somehow discouraging fact in the field of computational cancer research is that a voluminous literature of more than 150,000 papers documenting thousands of claimed biomarkers has been produced in medicine of which fewer than 100 have been validated for routine clinical practice \cite{Poste2011Bring}, and even fewer than 20 are recognized with variable levels of evidence in the 2014 European Society of Medical Oncology (ESMO) clinical practice guidelines for lung, breast, colon and prostate cancer \cite{Schneider2015Establishing}. Compared to the number of research findings in this area, the very few number of gene expression-based breast cancer prognostic signatures successfully implemented in clinical routines (Section \ref{sec1:molecular}) has inevitably raised controversies on the practical validity of molecular signatures. This is mainly because the vast majority of those findings are deficient in a proper validation procedure, not to mention validation oriented for clinical implication, resulting in an exaggeration of trivial findings and their clinical utility so that a large number of claimed signatures could very likely fail to add significantly incremental values assisting prognosis assessment and therapeutic decision making in addition to the use of standard clinico-pathological parameters. Notwithstanding, the research community generally holds an optimistic prospective towards the future, as long as proper validation pipelines will be taken systematically in all forthcoming research \cite{Michiels2016Statistical}. Since objectives accounting for clinical utility were not at all entailed in the first place and neither meta-analysis-based validation with multiple datasets nor cross-study validation was performed during the course of my doctoral studies, we cannot conclude with having identified any prognostic signatures. However, we will try our best to venture some caveats suggesting pitfalls in analyzing genomic data for breast cancer prognosis and biomarker discovery, in line with many previous attempts \cite{Ambroise2002Selection, Simon2003Pitfalls}:


* regarding prognosis improvement, a intriguing and anti-intuition observation by many previous studies \cite{Haury2011influence}, selecting a few features compared to whole genome often does not lead to drastic improvement as reported across all three studies in Section \ref{sec2:classification}, Section \ref{sec3:breastcancer} and Section \ref{sec4:performance}, while in the latter two sections it fortunately does not decrease significantly either... on the other hand, selecting with respect to given network compared to network-free often does not drastically improve the prognosis performance despite that features selected coherent with the given network indeed show superiority as in Section \ref{sec3:breastcancer} and Section \ref{sec4:performance}... In particular, it is not obvious that feature selection in genomic data analysis leads to the best-performing model for making the most accurate prognosis. Therefore, the assumption that a model based on only a few biomarkers rather than all genomic features available should better capture and explain the biological complexity related to cancer survival has not yet until now been definitively confirmed, and thus should be taken with caution.


* trust in the biological values of the biomarkers selected used in developing prognostic assays (or to further suggest therapeutic targets) should be taken with extreme caution... prognostic accuracy, cross-study reproducibility and functional interpretability of selected biomarkers are the least of indispensable requirements of candidate molecular signatures \cite{Haury2011influence}... the problem was quickly noticed from the beginning of computational cancer research when many claimed findings barely overlap such as 70-gene vs 76-gene only has 3 genes in common, plus later random genes associated to prognosis \cite{Venet2011Most}...


Our knowledge towards cancer biology is still far from complete but we are given the extraordinary opportunity in the era of big data to study cancer. Just bear in mind opportunities come with caveats that it calls for comprehensive study, proper validation as well as concerns such as clinical utility and cost-effectiveness for sake of true success...
