\chapter{Conclusion and Perspectives}
\label{chap:conclusion}

To summarize, the work presented in this thesis has been driven typically by the development and investigation of machine learning methods to address the computational challenges confronted in genomic data analysis: ranked-based approaches for improved breast cancer prognosis and network-guided approaches for enhanced biomarker discovery related to cancer survival. In fact, it is noteworthy that the theoretical and methodological contribution lies fundamentally in several branches of machine learning concerning applications across (but not limited to) cancer biology and social choice theory, specifically:

\paragraph{Learning with Rank Data.}

We have proposed two computationally attractive positive definite kernels between permutations, namely the Kendall and Mallows kernels, and further extended these kernels to rank data of complex structure that prevail in real-world applications including partial rankings, multivariate rankings and uncertain rankings (Chapter \ref{chap:kendall}). The significance of this work is of at least two folds: 
\begin{bulletList}
\item[1.] Thanks to these kernels, many kernel machines serve as off-the-shelf alternatives available to solve various problems pertaining to learning from rankings, or learning to rank, and can yield state-of-the-art performance that were demonstrated with an unsupervised cluster analysis of heterogeneous voting data and supervised biomedical classification tasks (Chapter \ref{chap:kendall}).
\item[2.] The Kendall embedding of the symmetric group brings forth novel incentives of learning on the symmetric group from unprecedented aspects. For instance, the Kendall embedding has motivated a geometric interpretation of the combinatorial problem of Kemeny aggregation based on which a tractable approximation bound was derived (Appendix \ref{chap:kemeny}) and can offer the opportunity of studying permutation problems such as seriation with yet another embedding following \cite{Fogel2013Convex, Lim2014Beyond}.
\end{bulletList}

\paragraph{Learning on Graphs.}

Given a network, we focused on network-guided feature selection coherent with the presumed network structure in two cases:
\begin{bulletList}
\item[1.] In case that the network is represented by an undirected graph and it encodes codependent relationships between features, assuming that neighboring features are encouraged to be selected simultaneously, we formalized the use of network-based wavelet smoothing as a regularization method for inducing structured sparsity with network-adaptive modularity in linear predictive models (Chapter \ref{chap:wavelet}).
\item[2.] In case that the network is represented by a directed graph and each circuit in it encodes the transduction of signal between features, assuming that circuit-level groups of features, if selected, are always selected simultaneously, we investigated the idea of first transforming the original representation of feature data into a circuit-level representation based on mathematical modeling of the network structure and then applying any standard feature selection algorithm which now becomes viable straightforwardly at the level of circuits (Chapter \ref{chap:hipathia}).
\end{bulletList}
Proof-of-methodology survival analysis of breast cancer was performed guided by a protein-protein interaction network (undirected) or a signaling pathway network (directed), and in both cases empirical superiority was demonstrated where biomarker discovery is enhanced by performing feature selection guided by a biological network as prior knowledge.



Research perspectives... bioinformatics ways to improve prognosis performance: subtyping of breast cancers and prognosis/treatment for each subtype as in commercial assays... integrative data analysis, multi-view multi-omics analysis, especially incorporating with clinical variables...


we do not claim to have identified prognostic biomarkers or signatures, the aim of the present thesis is instead to investigate ideas in designing biomarkers from a methodological point in machine learning... in fact, far from clinical implication... missing at least meta-analysis with multiple datasets or cross-study validation... A quite disappointing fact in the field of computational cancer research is that a voluminous literature of more than 150,000 papers documenting thousands of claimed biomarkers has been produced in medicine of which fewer than 100 have been validated for routine clinical practice \cite{Poste2011Bring}, and fewer than 20 are recognized with variable levels of evidence in the 2014 European Society of Medical Oncology (ESMO) clinical practice guidelines for lung, breast, colon and prostate cancer \cite{Schneider2015Establishing}. The very few number of gene expression-based breast cancer prognostic predictors successfully implemented in clinical routines compared to the number of research findings in this area raises controversies on the practical validity of molecular signatures... mainly because of lack of proper validation of the computational findings for clinical implication, resulting in overly exaggeration of trivial findings and clinical utility where a large number of identified signatures failed to add significant values involving prognosis and therapeutic decision making that are incremental to the use of conventional clinico-pathological parameters alone \cite{Michiels2016Statistical}... In the end, as we have not at all systematically considered clinical-oriented purposes, we cannot conclude with any contribution on clinically meaningful biomarkers, however we can give some caveats suggesting pitfalls in biomarker discovery on top of \cite{Ambroise2002Selection, Simon2003Pitfalls}...

* regarding prognosis improvement, a intriguing and anti-intuition observation by many previous studies \cite{Haury2011influence}, selecting a few features compared to whole genome often does not lead to drastic improvement as reported across all three studies in Section \ref{sec2:classification}, Section \ref{sec3:breastcancer} and Section \ref{sec4:performance}, while in the latter two sections it fortunately does not decrease significantly either... on the other hand, selecting with respect to given network compared to network-free often does not drastically improve the prognosis performance despite that features selected coherent with the given network indeed show superiority as in Section \ref{sec3:breastcancer} and Section \ref{sec4:performance}... In particular, it is not obvious that feature selection in genomic data analysis leads to the best-performing model for making the most accurate prognosis. Therefore, the assumption that a model based on only a few biomarkers rather than all genomic features available should better capture and explain the biological complexity related to cancer survival has not yet until now been definitively confirmed, and thus should be taken with caution.

* trust in the biological values of the biomarkers selected used in developing prognostic assays (or to further suggest therapeutic targets) should be taken with extreme caution... prognostic accuracy, cross-study reproducibility and functional interpretability of selected biomarkers are the least of indispensable requirements of candidate molecular signatures \cite{Haury2011influence}... the problem was quickly noticed from the beginning of computational cancer research when many claimed findings barely overlap such as 70-gene vs 76-gene only has 3 genes in common, plus later random genes associated to prognosis \cite{Venet2011Most}...

Our knowledge towards cancer biology is still far from complete but we are given the extraordinary opportunity in the era of big data to study cancer. Just bear in mind opportunities come with caveats that it calls for comprehensive study, proper validation as well as concerns such as clinical utility and cost-effectiveness for sake of true success...
