\chapter{Introduction}
\label{chap:intro}

\begin{chapabstract}

\textrm{{\bf Abstract:}} This thesis work has been driven typically by the development and investigation of machine learning methods in genomic data analysis for improved breast cancer survival prediction and for detecting robust molecular signatures related to cancer prognosis. Towards these goals, two directions in terms of methodologies have been explored for supervised survival prediction and structured feature selection, which are rank-based and network-guided approaches. This chapter serves as a general introduction to the thesis that briefly summarizes the background of breast cancer prognosis and summarizes the motivation of my thesis research to address the computational challenges in genomic data analysis for cancer prognosis.
\linebreak
\vskip 0.1in
\noindent \textrm{{\bf Résumé:}}

\end{chapabstract}

\section{General Background}

Breast cancer refers to a malignant tumor that has developed from cells in the breast. Uncontrolled growth of cancer cells can invade nearby healthy breast tissue over time, and if cancer cells get into the lymph nodes that are small organs that filter out foreign substances in the body, they could then have a system of spreading further into other parts of the body and form new tumors in distant organs or tissues, a process called distant metastasis that aggravates the situation to a significant extent. Breast cancer is the most common cancer in women worldwide and second most common cancer overall for both genders in terms of incidence rates (after lung cancer). It is one of the leading causes of women's death from cancer, and over 508,000 women worldwide were estimated to have died in 2011 due to breast cancer (Global Health Estimates, WHO 2013). Survival rates have in general been improving over the past decade as a result of increased awareness, earlier detection through screening, adequate medical care and cancer treatment advances, with the caveat that rates vary greatly worldwide and still remain quite low in less developed countries.



If diagnosed at an early stage, the initial treatment for breast cancer is usually accomplished by complete removal of tumor by surgery or radiation (without damage to the rest of the body). After the initial treatment (or in case that the initial treatment should not be applicable), many patients are advised to receive systemic treatment including adjuvant chemotherapy, hormone therapy and targeted therapy to lower the risk of relapse that is the recurrence of cancerous conditions and/or to prevent metastasis. The decision of receiving systemic treatment is made based on prognosis of the cancer patient, the estimation of the risk of relapse or likely course of outcome if no systemic treatment is given after the initial treatment.\footnote{In fact, two questions need to be addressed in decision making for cancer treatment: prognosis that is the identification of those patients who are most likely to benefit from the systemic treatment, and prediction that is the determination on which specific treatment should be most responsive and effective for the patients. While prognosis and prediction are equally important and usually discussed together in literature, prediction will be omitted from discussion for ease of presentation of this thesis.} For example, as the most common type of systemic treatment, adjuvant chemotherapy usually involves cytotoxic drugs and has strong deleterious side effects. Consequently, such aggressive treatments are most beneficial for patients with poor prognosis but the intake of adjuvant chemotherapy should be minimized for those that do not necessarily benefit much from it. Therefore, accurate prognosis for each individual patient is of chief importance in order to personalize the best therapeutic option.



conventionally prognosis of breast cancer is based solely on clinico-pathological features... Several well-established prognostic factors are commonly used to guide the clinical management of breast cancer (ref table)... a few words of subtypes and treatment/survival... well-known 6 prognostic factors used by Adjuvant! Online\footnote{www.adjuvantonline.com} : patient age, tumor size, grade, hormone receptor status, number of positive lymph nodes and comorbidity level... 


Table: Clinico-pathological features commonly used as prognostic factors of breast cancer. (saini's thesis table 1.1, pia's thesis table 1)...



Due to complexity and diversity of breast cancer, however, patients of similar clinico-pathological type can still have remarkably different prognosis... according to a study in \cite{Veer2008Enabling}, in the US approximately up to 70\% of breast cancer patients with negative lymph nodes and a tumour size 1 to 2 cm undergo a toxic therapy that they would very likely not benefit from after they were actually already cured of their disease... indicates that the clinico-pathological information alone is far from sufficient to characterize the outcome of each diseased individual. Difficulties in detecting reliable and efficient prognostic factors for breast cancer remains in the biological heterogeneity across cancer tumors. it is challenging to accurately do prognosis or identify prognostic factors for each particular case... 






\section{Genomic Data Analysis for Cancer Prognosis: Prospects and Challenges}




among many theories of cancer, one theory states breast cancer is always caused by a genomic abnormality.. It has been long established that genomic features contain unique characteristics of each patient and offer the opportunity of scrutinizing the individuality of each breast tumor. Often termed by \textit{biomarker} are such molecular-level prognostic factors that characterize intrinsic biological heterogeneity of tumours.  The characterization of such biomarkers for breast cancer prognosis include gene expression, SNPs, etc... many have been associated with breast cancer survival (ref table)...


Table: Some important genes whose mutation carriers are associated with high risk for breast cancer. (Pia's thesis table 2)


The advancement of genomic profiling technologies in the past decades has been motivating the trend of using genomic information in decision-making regarding prognosis improvement, as millions of genomic features can be efficiently collected from patients and tumors. Take as an example, high-throughput gene expression profiling technologies: DNA microarray and RNA-seq, NGS, WGS...



Genomic data analysis is a broad topic... In the genre of genomic data analysis for cancer prognosis, two major questions need to be addressed: 


1) survival risk prediction via supervised statistical inference...


prognosis in the language of machine learning - classification or survival analysis... four types of survival : DMFS, RFS, DFS, OS (saini's thesis table 1.1); binary labels converted from survival... feature selection (2nd aspect below) or not (whole genome)...



2) biomarker discovery via feature selection...

Promising results with a few genomic tests available... Mammaprint \cite{Veer2008Enabling} and other tests http://www.breastcancer.org/symptoms/testing/types/mammaprint






Despite the successful stories, it has been widely recognized as a very challenging problem to extract potentially valuable information from genomic data. Besides the complexity and heterogeneity of the disease, relatively small number of clinical samples, the inherent measurement noise in high-throughput experimental data, the heterogeneity of patient data makes it demanding... furthermore, robustness of the identified biomarkers and \textit{a posteriori} interpretation of the findings are also major concerns... literature review???






\section{Contribution of the Thesis}

This thesis is conceived following the two lines of ideas of performing genomic data analysis for cancer prognosis and will be outlined by projects, each presented in one chapter.




\paragraph{Rank-based Approaches for Improved Survival Risk Prediction.}

The first line of ideas is to perform gene expression data analysis based solely on the ranking of the expression levels of multiple genes while their real-valued measurements are disregarded. It is based on the assumption that the relative ordering of gene expression levels can be potentially more informative than their real-valued measurements when used in some biomedical applications since high-throughput high-dimensional genomic data are often subject to high measurement noise. From the point view of machine learning, it reduces to the study of a particular type of structured data, specifically rankings. It is well-known that kernel methods have found many successful applications where the input data are discrete or structured including strings and graphs. The first project of my doctoral studies was focused on proposing computationally attractive kernels for rank data and applying kernel methods to problems involving rankings. Central to this work was the observation that the widely used Kendall tau correlation and the Mallows similarity measures are indeed computationally attractive positive definite kernels for rankings. These kernels were further tailored for more complex types of rank data that prevail in real-world applications, especially when rankings come from real-valued vectors by keeping simply the relative ordering of the values of multiple features thereof. Thanks to these kernels, many off-the-shelf kernel machines are available to solve various problems at hand. It is worth special mention that, despite that the project was initially motivated by biomedical applications, the prospective contribution of this work concerns applications from many fields of machine learning pertaining to learning from rankings, or learning to rank. This study will be presented in Chapter \ref{chap:kendall}.



The study of the Kendall kernel for rankings has paved an unprecedented way towards a deeper understanding of a classical problem called Kemeny aggregation from the field of Social Choice Theory. Kemeny aggregation searches for a consensus ranking that best represents a collection of individual rankings in the sense that the sum of the Kendall tau distance between each ranking and the consensus is minimized. Kemeny aggregation is often considered to provide the ideal solution among all ranking aggregation criteria but the Kemeny consensus is known to be NP-hard to find. Many tractable approximations to the Kemeny consensus have therefore been proposed and well studied. Since the Kendall kernel derives from an inner product of a Euclidean space, the Kendall tau distance derives from a squared Euclidean distance and thus the combinatorial problem of Kemeny aggregation is endowed with a novel intuitive interpretation from a geometric point of view. Based on this observation, a tractable upper bound of the estimation error in terms of the distance between the Kemeny consensus and its approximation is established. This upper bound requires little assumption on the approximation procedure or the collection of rankings to aggregate. However, due to its remote connection to cancer prognosis that is the primary objective of the present thesis, this study will be presented in Appendix \ref{chap:kemeny}. 



\paragraph{Network-guided Approaches for Enhanced Biomarker Discovery.}


The second line of ideas of performing genomic data analysis for cancer prognosis is to consult biological networks as domain-specific knowledge to guide our analysis. In fact, biological networks are a common way of depicting functional relationships between genes that have been accumulated from decades of biomedical research, and they can be particularly valuable when incorporated as prior knowledge in order to improve the selection efficacy of prognostic biomarkers related to breast cancer prognosis. Typical examples of biological networks include protein-protein interaction network and signaling pathway network.


Two projects were initiated on network-guided gene expression data analysis for breast cancer prognosis and for identification of genomic signatures related to cancer survival. In one project, we focused on the study of structured regularization in generalized linear models for selecting prognostic genes where the regularization method is designed so that genes closer on the biological network are encouraged to be selected simultaneously. In fact, in order to achieve such simultaneous modularity and sparsity coherent with respect to presumed network structure, a popular method called network-based wavelet smoothing has been successfully applied in many applications from the field of signal processing. Therefore, we were motivated to investigate the potential of this method in biomedical applications, using a breast cancer gene expression dataset and a protein-protein interaction network to guide our analysis albeit the methodology is generally applicable to various types of genomic data and biological networks. In particular, the method allows to designate a few prognostic gene modules with intra-collaborative functionality rendering readily interpretable insights potentially related to cancer survival. Numerical results demonstrated that, compared to several network-free and some established network-based regularization methods, network-based wavelet smoothing was able to improve the selection efficacy of prognostic genes in terms of stability, connectivity and interpretability, while achieving competitive performance of survival risk prediction for breast cancer. This study will be presented in Chapter \ref{chap:wavelet}.


In another project, we focused on a particular type of biological network namely signaling pathway network. Based on a modeling framework of cell signaling, gene expression profiles can be translated into personalized profiles of signaling pathway activities by integrating signaling pathway network. When gene-level profiles are replaced by these derived pathway-level profiles as input to many off-the-shelf computational tools, a simple scheme emerges where gene-level analysis is easily promoted to pathway-level analysis. Analysis made at the level of functional pathways consequently leads to better understanding in terms of biological relevance. Notably, when combined with feature selection oriented algorithms, the proposed scheme enables direct identification of pathway-level mechanistic signatures as an alternative to conventional gene-based signatures, which provides more informative insights into the cellular functions involved in cancer survival. This study will be presented in Chapter \ref{chap:hipathia}.




\paragraph{Other Contribution.}


During the course of my doctoral studies, I have undertaken some other projects as well. In 2013, Elsa Bernard, Erwan Scornet, V\'{e}ronique Stoven, Thomas Walter, Jean-Philippe Vert and I from our laboratory participated as a team in an international bioinformatics competition called \textit{DREAM 8 NIEHS--NCATS--UNC Toxicogenetics Challenge}. In the competition, participants were asked to predict the response of human cell lines exposed to various toxic chemical compounds, based on the chemicals' molecular characterization and the cell lines' transcriptome. Finally our team won second place with a kernel bilinear regression model. Oral presentation was accepted to \textit{NIPS Workshop on Machine Learning in Computational Biology (MLCB)} and later invited to \textit{RECOMB Conference on Regulatory and Systems Genomics}. This work has been published as part of the crowd-source collaboration as a result of the competition \cite{Eduati2015Prediction} but excluded from the present thesis.



During the spring of 2015 when I was interning at Roche Diagnostics, Penzberg, Germany, I worked on failure state prediction for automated analyzers for analyzing biological samples, and a European patent application regarding the methodology was filed by Roche in December 2016 and is currently pending approval \cite{Jiao2016Failure}. Due to corporate confidentiality issues, this study will not be included in the present thesis.

