\chapter{Introduction}
\label{chap:intro}

Breast cancer is one of the most common malignancy for women and breast cancer prognosis is currently one of the leading problems of great interest. The advent of high-throughput platforms for gene expression profiling motivates the popular trend of using molecular biomarkers for clinical decision making regarding prognosis improvement assisting therapeutic strategies. It has been widely recognized as a very promising yet challenging problem to extract potentially valuable information from genomic data. The relatively small number of clinical samples, the inherent measurement noise in high-throughput experimental data and the heterogeneity across patients make it demanding to detect reliable biomarkers.

MammaPrint \cite{VantVeer2008Enabling}.

An important feature of the proposed scheme is the idea of promoting gene-level analysis to pathway-level analysis by applying the \textit{hiPathia} method to obtain patient-specific personalized profiles of signaling circuit activity. Over the past decade, a great deal of efforts have been devoted to analyzing gene expression data in extensively diverse biomedical applications revolving around cancer as well as other diseases. Typical examples include cancer prognosis and subtyping, patient stratification and biomarker detection. Depending on the specific problem at hand, many existing algorithms that were originally designed for gene expression profiles can be easily adapted to signaling circuit activity profiles. Consequently the analysis is made at the level of functional pathways instead of individual genes, often leading to better understanding in terms of biological relevance.

My thesis research has been driven typically by the development and investigation of machine learning methods to detect robust molecular signatures in genomic data for improved cancer prognosis. Towards this goal, two directions in methodology were taken for predictive modeling and structured feature selection. In this statement, I briefly summarize my past and current research, and also lay out some future directions.

The first line of ideas was to perform gene expression data analysis based solely on the ranking of the expression levels of multiple genes while their real-valued measurements are disregarded. It is based on the assumption that the relative ordering of gene expression levels can be more informative than their real-valued measurements when used in some biomedical applications since high-throughput high-dimensional genomic data are often subject to high noise. Hence the majority of my thesis work has been focused on the study of a particular type of structured data, specifically rankings. It is well-known that kernel methods have found many successful applications where the input data are discrete or structured including strings and graphs. However, surprisingly little attention has been paid to applying kernel methods to problems involving rankings, due to the lack of computationally attractive kernels for such data. Central to this work was the observation that the widely used Kendall and Mallows similarity measures are positive definite kernels between rankings. These two kernels were further extended for more complex types of rank data that prevail in real-world applications. Thanks to these kernels, many kernel machines are readily available to solve various problems of learning from rankings or learning to rank. Numerical experiments demonstrated that kernel methods with the proposed kernels achieve state-of-the-art performance on tasks such as clustering heterogeneous rank data and high-dimensional classification problems in biomedical applications. This work was first published in the \textit{International Conference on Machine Learning (ICML)} \cite{Jiao2015Kendall} and now under revision for the journal \textit{IEEE Transactions on Pattern Analysis and Machine Intelligence (TPAMI)}. Further, I have developed an \texttt{R} package that implements kernel functions and kernel methods for analysis of various types of rank data.

There are many interesting extensions regarding the Kendall and Mallows kernels that are left to future investigation. For instance, one direction would be to introduce optional weights to the proposed kernels such that the reverse ordering of different item pairs can contribute with unequal importance to the kernel evaluation. This can be particularly interesting if changing the rank of a top-positioned item should result in a higher penalty than changing the rank of a bottom-positioned item when evaluating similarity between rankings. Another direction would be to include high-order comparisons in measuring the similarity between rankings. Since the fast computation of the Kendall and Mallows kernels is balanced by the fact that they only rely on pairwise statistics between items given to rank, computationally tractable extensions including higher-order statistics, such as three-way comparisons, could potentially enhance the discriminative power of the proposed kernels. A third direction would be to extend the proposed kernels to rankings on a partially ordered set. In fact, the previous work lies on the implicit assumption that the item set given to rank is naturally associated with a total order such that the proposed kernels are defined by comparing all item pairs. On one hand, the proposed kernels would be ill-defined when some item pairs were conceptually incomparable. On the other hand, in case that a total order does exist, it remains unclear whether reducing to a subset of pairwise comparisons could outperform the originally defined kernels for specific applications.

The second line of ideas was to consult biological networks as domain-specific knowledge to guide genomic data analysis. In fact, biological networks are a common way of depicting functional relationships between genes that have been accumulated from decades of biomedical research, and they can be particularly valuable when incorporated as prior knowledge in order to improve the performance of survival risk prediction and the efficacy of biomarker selection related to breast cancer prognosis. Two projects were initiated to explore the potential of network-guided gene expression data analysis and biomarker detection. In the first project, based on a modeling framework of cell signaling that translates gene expression profiles into personalized profiles of signaling pathway activities, we proposed such a simple scheme by integrating a signaling pathway network into gene expression data analysis that gene-level analysis is easily promoted to pathway-level analysis. When combined with state-of-the-art computational tools, the scheme enables the identification of pathway-level mechanistic biomarkers as an alternative to conventional gene-based biomarkers, which provides more informative insights into cellular functions involved in cancer mechanism. This work was initiated when I was visiting Centro de investigaci\'{o}n Pr\'{i}ncipe Felipe (CIPF), Valencia, Spain, and has been submitted to the journal \textit{Bioinformatics}. In the second project, we studied network-based wavelet smoothing for analysis of genomic data assisted by a protein-protein interaction network, which is methodologically a class of penalty terms for regularizing generalized linear models in order to achieve structured modularity-sparsity for the coefficient parameter. Consequently, the learned coefficient parameter of the linear model will designate a few prognostic subnetworks consisting of genes with collaborative functionality involved in cancer biology. Results demonstrated that, compared to several network-free and other network-based regularization methods, network-based wavelet smoothing was able to improve the efficacy of gene selection in terms of stability, connectivity and interpretability, while achieving competitive performance of survival risk prediction for breast cancer prognosis. A journal submission on this work is under preparation.

During the course of my doctoral studies, I have undertaken some other projects as well. Some colleagues and I participated in the \textit{DREAM 8 NIEHS--NCATS--UNC Toxicogenetics Challenge}, an international bioinformatics competition on predicting human population responses to toxic compounds, and we won second place with a kernel bilinear regression model. Oral presentation was accepted to \textit{NIPS Workshop on Machine Learning in Computational Biology (MLCB)} and later invited to \textit{RECOMB Conference on Regulatory and Systems Genomics}. While this work has been published as part of the competition-based collaboration in Nature Biotechnology \cite{Eduati2015Prediction}, a journal submission with a particular emphasis on our methodology is underway to a special issue in \textit{Molecular Informatics}. When I was interning at Roche Diagnostics, Penzberg, Germany, I worked on failure state prediction for automated analyzers for analyzing biological samples, and a European patent application regarding the methodology was filed by Roche in December 2016. A project on a theoretical machine learning topic was towards a deeper understanding of a classical problem in social choice theory called Kemeny aggregation. We endowed this combinatorial problem with a novel and intuitive interpretation from a geometric point of view, which we believe will pave an unprecedented way to addressing some fundamental questions. This work stemmed from discussion with a PhD student after my presentation at ICML on the work of the Kendall kernel, and resulted in another ICML publication the next year upon collaboration \cite{Jiao2016Controlling}.
